File: lab11Calificado/Lab11_PLANTILLA/TestSoftSolution/TestSoftBusiness/src/main/java/pe/edu/pucp/testsoft/business/bo/IBaseBO.java
package pe.edu.pucp.testsoft.business.bo;

import java.util.List;

// BO genérico. Lanza Exception: la capa de negocio valida ANTES de delegar al DAO.
public interface IBaseBO<T> {
    int insertar(T objeto) throws Exception;
    int modificar(T objeto) throws Exception;
    int eliminar(int id) throws Exception;
    T buscarPorId(int id) throws Exception;
    List<T> listarTodos() throws Exception;
}

File: lab11Calificado/Lab11_PLANTILLA/ExamenParcialCSharp/GestionAlumnosBusiness/BO/IBaseBO.cs
namespace GestionAlumnosBusiness.BO
{
    public interface IBaseBO<T>
    {
        int Insertar(T objeto);
        int Modificar(T objeto);
        int Eliminar(int id);
        T BuscarPorId(int id);
        List<T> ListarTodos();
    }
}
